### PR TITLE
Src/extras/core:   [remake] move to es6 classes

### PR DESCRIPTION
**Pull request by DefinitelyMaybe**
Related issue: #20143

**Description**

Remake of related issue.

PR also includes change to `.gitignore`, which added some .ps1 scripts when I ran `npm i --prefix test`

### CODE REVIEW

**Comment by DefinitelyMaybe** [comment]
resolved conflicts from previous PR

**Comment by Mugen87** [comment]
I think if you mark your ES6 class PRs as `Draft PRs`. Most of these can't be merged without adjusting the respective example code.

**Comment by DefinitelyMaybe** [comment]
I was moved to draft earlier because things weren't checking out but, yes, should've done that in the first instance after initial PR tests failed.

What was the respective example code that needed to be changed for this PR? Could I simply include that adjustment within this PR?

File: .gitignore [hunk 1]
```diff
@@ -17,7 +17,8 @@ test/*
 !test/*/
 !test/*.*
 test/*.cmd
+test/*.ps1
```
Comment (mrdoob) [inline .gitignore:20]: What is this?

File: src/extras/core/Font.js [hunk 2]
```diff
@@ -27,116 +27,116 @@ class Font {
 
 	}
 
-}
+	createPaths( text, size, data ) {
 
-function createPaths( text, size, data ) {
```
Comment (mrdoob) [inline src/extras/core/Font.js:32]: Why not keep this as a function outside the class?

**Comment by mrdoob** [comment]
This PR seems to be almost good to go. The only issue is the `createPaths` change.

**Comment by DefinitelyMaybe** [comment]
This PR is too old. Someone else has already changed `font.js` to a class and left `function createPaths(...` outside the class. The strangeness from `test/*.ps1` is from new commits.

I will remake again with the other changes I made.

File: .gitignore (hunk 1 above)
Comment (looeee) [reply .gitignore:20]: Windows PowerShell scripts.

**Pull request closed without merging**